src/views/Shopping/Shopping.test.jsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import Shopping from './Shopping';
+import userEvent from '@testing-library/user-event';
+import { UserProvider } from '../../context/UserCtx';
+import { render, screen } from '@testing-library/react';
+
+it('allows user to add, update, and delete items', async () => {
+    const {container} = render(
+        <UserProvider>
+            <Shopping />
+        </UserProvider>
+    );
+
+    const userInput = screen.getByLabelText(/name-input/i);
+    const logBtn = screen.getByLabelText(/log-button/i);
+
+    userEvent.type(userInput, 'Izzie');
+    userEvent.click(logBtn);
+
+    await screen.findAllByText(/Izzie/i);
+
+    const nameInput = await screen.findByLabelText(/new item input/i);
+    const submitBtn = await screen.findByLabelText(/submit new item/i);
+
+    userEvent.type(nameInput, 'name');
+    userEvent.click(submitBtn);
+
+    const firstItem = await screen.findByText(/name/i);

```

Comment:
Great job! This is definitely how you should be doing the behavior tests. Check for the static element, make an input/change, check for the updated value the change created. :) 

You are good to go ahead and merge this one now! 💯